### PR TITLE
Increase Minimum Line Code Coverage Threshold

### PR DESCRIPTION
**Pull request by szh**
### Desired Outcome

The current code coverage is currently hovering around 65%. Try to get it above 70% and increase the minimum coverage to 70% in the Jenkinsfile.

Min. Coverage: 70%

Desired Coverage: 80%

Optimal Coverage: 90%

### Implemented Changes

- Removed empty_test.go files from mock directories, so those directories are not counted towards coverage. This increase combined coverage from UT and integration tests to 68.7%.
- Added more unit tests to increase combined coverage to 70.7%.
- Modified Jenkinsfile to fail if combined coverage falls below 70%.
 
### Connected Issue/Story

CyberArk internal issue link: [ONYX-14781](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-14781)

### Definition of Done

- [x] Cobertura reports at least 70% code coverage
- [x] CI fails if test coverage is less than 70%

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [x] This PR includes new unit and integration tests to go with the code
  changes, or
- [ ] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

**Comment by adamouamani** [comment]
Looks good

File: internal/configurationmanagers/configfile/fs_watcher.go [hunk 1]
```diff
@@ -7,11 +7,13 @@ import (
 	"github.com/fsnotify/fsnotify"
 )
 
+var logFatal = log.Fatal
```
Comment (rpothier) [inline internal/configurationmanagers/configfile/fs_watcher.go:10]: Why was this added?

File: pkg/k8sclient/clientset/versioned/fake/empty_test.go [hunk 2]
```diff
@@ -1,6 +0,0 @@
-// This is an empty test file used to force Go test to include this package in
```
Comment (rpothier) [inline pkg/k8sclient/clientset/versioned/fake/empty_test.go:1]: Weren't these added to make the test reporting accurate?

Comment (rpothier) [inline pkg/k8sclient/clientset/versioned/fake/empty_test.go:1]: Weren't these added to make the test reporting accurate?

**Review by rpothier — COMMENTED** [review]
Looks good, just a couple questions.

File: internal/configurationmanagers/configfile/fs_watcher.go (hunk 1 above)
Comment (szh) [reply internal/configurationmanagers/configfile/fs_watcher.go:10]: This is so we can mock the calls to logFatal without actually stopping execution.

File: pkg/k8sclient/clientset/versioned/fake/empty_test.go (hunk 2 above)
Comment (szh) [reply pkg/k8sclient/clientset/versioned/fake/empty_test.go:1]: Yes, but they were added even to directories that shouldn't need to be tested, such as directory which only contains generated code.

https://github.com/cyberark/secretless-broker/blob/92983c5fbd0fcb992c340954efb694afc00b64a9/pkg/k8sclient/clientset/versioned/fake/doc.go#L17

**Review by rpothier — APPROVED** [review]
LGTM!
